gnucash/gnome/window-reconcile.c

```diff
@@ -384,9 +387,51 @@ gnc_start_recn_date_changed (GtkWidget *widget, startRecnWindowData *data)
     gnc_numeric new_balance;
     time64 new_date;
 
+    gboolean show_warning = FALSE;
+    GtkWidget *future_icon, *future_text;
+    gint days_after_today;
+
+    new_date = gnc_date_edit_get_date_end (gde);
+
+    days_after_today = round ((gnc_time64_get_day_end (new_date) -
+                               gnc_time64_get_today_end ()) / 86400);
```

Comment:
`round()` is for floating-point. This compiles only because the `int64_t` is implicitly cast to a `double`, rounded, and then the returned double is implicitly cast back to an `int64_t`.

In C integer division always returns an integer of the same type as the larger operand and the remainder is always discarded. The `%` operator returns the remainder, but in general there won't be any. The exception would be crossing a DST boundary, where going from standard time to daylight time will lose a whole day because the interval is less than 86400 seconds. To deal with that case I suggest adding 3600 to the difference before dividing.

Also, it's bad form to stick a raw constant into calculation. Instead add a constant decl to the function's declaration block (or put at the top of the file if you need the constant in more than one place):
```
static const time64 secs_per_day = 86400;
static const time64 secs_per_hour = 3600;
...
/* Add secs_per_hour to the difference to compensate for the short day when
 * transitioning from standard to daylight time.
 */
days_after_today = (gnc_time64_get_day_end (new_date) - 
                                    gnc_time64_get_today_end + secs_per_hour) / secs_per_day;
```

Reply:
the secs_adjustment may be up to 3hours sometimes... see Brazil in https://en.wikipedia.org/wiki/Daylight_saving_time, so, to be very general I think best adjust by `secs_per_day/2`

Reply:
Read that more carefully: It says "In early 2008, central Brazil was one, two, or three hours ahead of the eastern United States, depending on the date." That's because Brazil's DST wasn't (and probably isn't) aligned with the US's. Fortunately for Brazilians it doesn't mean that a DST shift in Brazil can be 3 hours in a month's time.

We're not looking for TZ shifts here either: The two times are generated within a few microseconds of each other and aside from Dr. Who nobody's changing timezones in a fraction of a second.

Reply:
Ok :) this is very much small print.